src/dancefloor.js

```diff
@@ -15,17 +15,34 @@ export default class Dancefloor extends React.Component {
             }, 100); // css transition delay
             let dx = 0;
             let dy = 0;
-            switch (evt.keyIdentifier.toLowerCase()) {
-                case 'up':
-                    dy += 10; break;
-                case 'down':
-                    dy -= 10; break;
-                case 'left':
-                    dx -= 10; break;
-                case 'right':
-                    dx += 10; break;
-                default:
-                    break;
+            if (evt.keyIdentifier !== undefined) {
```

Comment:
ah ha!  i see. i thought the event was fake React generated event, so it would be consistent.  originally i was going to use the react event, but it wasn't consistent enough.  looks like i forgot to adjust for browser events.  i like what you've done here.  i did some reading on this--looks like you did too. key events are wonky.  we will still miss a few mainstream browsers that aren't up to do with both of our strategies.  how about this, to catch 'em all, and catch all browsers?

```js
let key = evt.key || evt.keyIdentifier || evt.keyCode.toString();
switch (key.toLowerCase()) {
    // 37 <, 38 ^, 39 dwn, 40 >
    case '37':
    case 'left':
        dx -= 10; break;
    case '38':
    case 'up':
        dy += 10; break;
    case '39':
    case 'down':
        dy -= 10; break;
    case '40':
    case 'right':
        dx += 10; break;
    default:
        break;
...
```